addons/test_mail_sms/tests/test_sms_composer.py

```diff
@@ -133,6 +133,22 @@ def test_composer_comment_nofield(self):
             sms_composer._action_send_sms()
         self.assertSMSNotification([{'number': self.random_numbers_san[0]}], self._test_body)
 
+    def test_composer_comment_invalid_field(self):
+        """ Test the Send Message in SMS Composer when a Model does not contain a number field name """
+        test_record = self.env['mail.test.sms.partner'].create({
+            'name': 'Test',
+            'customer_id': self.partner_1.id,
+        })
+        sms_composer = self.env['sms.composer'].create({
+            'body': self._test_body,
+            'number_field_name': self.random_numbers_san[0],
+            'res_id': test_record.id,
+            'res_model': 'mail.test.sms.partner'
+        })
+        with self.mockSMSGateway():
+            sms_composer._action_send_sms()
+        self.assertSMSNotification([{'partner': test_record.customer_id}], self._test_body)
+
```

Comment:
It doesn't seem like the test fails prior to your fix. Always make sure that's the case, it also helps make sure your changes make sense and are valid.

You need to either set `number_field_name` in the composer manually to some field that does not exist (and check that it does not exist, for completeness)

Or patch `_sms_get_number_fields` for `mail.test.sms.partner` so that it returns a non-existing field, which matches the flow better but requires patching.

Your choice.

Reply:
Not critical, but can you also check that phone_nbr isn't a field with just `assertNotIn(sms_composer._fields, 'phone_nbr')`?

One never knows..

Reply:
I had to use ``','.join(test_record._fields)`` because ``assertNotIn()`` waits for a string on the left when the right is also a  string